absl/strings/str_cat.cc

```diff
@@ -89,7 +89,7 @@ static char* Append(char* out, const AlphaNum& x) {
   // memcpy is allowed to overwrite arbitrary memory, so doing this after the
   // call would force an extra fetch of x.size().
   char* after = out + x.size();
-  memcpy(out, x.data(), x.size());
+  if (x.size()) memcpy(out, x.data(), x.size());
```

Comment:
Can you add a test that would hit this behavior?

Reply:
I can easily demonstrate with  -fsanitize=undefined ` flag and in that case ubsan just terminates the process. I am not sure it's possible to uncover the bug within the current capabilities of gtest framework without ubsan.

Reply:
You don't have to demonstrate that the process gets terminated. In fact, you want the opposite. I'd like to see something that demonstrates that the process doesn't get terminated under the conditions in this bug. If someone ever removes the branch and runs the test under ubsan, I'd like to know something is wrong.

Reply:
Hey Derek, I am sorry, I do not understand what you mean. The bug is that with specific compiler options UB sanitizer terminates the process. I am not aware of any other manifestation of the bug.
To demonstrate it all you need is to call `StrCat(string_view{})`.